cpp/include/cudf_test/column_wrapper.hpp

```diff
@@ -202,6 +208,46 @@ rmm::device_buffer make_elements(InputIterator begin, InputIterator end)
   return rmm::device_buffer{elements.data(), size * sizeof(RepType), rmm::cuda_stream_default};
 }
 
+#ifdef __CUDACC__
+struct to_rep {
+  template <typename T>
+  __device__ typename T::rep operator()(T fp)
+  {
+    return fp.value();
+  }
+};
+
+template <typename ElementTo, typename ElementFrom, typename InputIterator>
+rmm::device_buffer make_device_elements(InputIterator begin, InputIterator end)
```

Comment:
I tried to think of ways to simplify this function, which has a decent amount of repeated logic. I did not come up with anything that I strongly prefer over the current state.

Reply:
To make `thrust::copy` as common code, `RepType` would need 2 `std::conditional`, which sacrifices readability.
so, left it like this.
Also, host iterator functions could be combined like this too.

Reply:
Awesome suggestion. Might need this tweak to handle the last case when `is_fixed_point<ElementFrom>()`?
```
    using Transformer = std::conditional_t<is_fixed_point<ElementFrom>(),
                                           to_rep,
                                           fixed_width_type_converter<ElementFrom, ElementTo>>;
    auto transform_begin = thrust::make_transform_iterator(begin, Transformer{});
```

Reply:
There is one problem. `ElementTo::rep` in std::conditional_t is not lazy evaluated.
So, it fails to compile.
With some machinery, this can be made to work.

```cpp
// TODO replace with std::type_identity in C++20
template <class T>
struct type_identity {
  using type = T;
};

template <class T>
struct rep_type {
  using type = typename T::rep;
};
  using RepType = typename std::conditional_t<cudf::is_fixed_point<ElementTo>(),
                                              rep_type<ElementTo>,
                                              type_identity<ElementTo>>::type;
```